layers/fuzz_layer.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 The GoPacket Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license that can be found
+// in the LICENSE file in the root of the source tree.
+
+package layers
+
+import (
+	"encoding/binary"
+
+	"github.com/google/gopacket"
+)
+
+// FuzzLayer is a fuzz target for the layers package of gopacket
+// A fuzz target is a function processing a binary blob (byte slice)
+// The process here is to interpret this data as a packet, and print the layers contents.
+// The decoding options and the starting layer are encoded in the first bytes.
+// The function returns 1 if this is a valid packet (no error layer)
+func FuzzLayer(data []byte) int {
+	if len(data) < 3 {
+		return 0
+	}
+	// use the first two bytes to choose the top level layer
+	startLayer := binary.BigEndian.Uint16(data[:2])
+	var fuzzOpts = gopacket.DecodeOptions{
+		Lazy:                     data[2]&0x1 != 0,
+		NoCopy:                   data[2]&0x2 != 0,
+		SkipDecodeRecovery:       data[2]&0x4 != 0,
```

Comment:
If SkipDecodeRecovery is false (as it would be ~half of the time), then if a layer panics, [gopacket.NewPacket will recover](https://github.com/google/gopacket/blob/1d829e51f0c85294eeedb06477a6d369fb5be0ea/packet.go#L197-L201), and thus this crash will be hidden from the fuzzer.

Is it worth setting SkipDecodeRecovery to true to avoid this from happening?

Reply:
Indeed.

But when SkipDecodeRecovery is false, the fuzz target can still time out, or crash while calling `gopacket.LayerString`

So, it seems worth to keep it, even if it will not find most crashes